internal/services/cognitive/azure_ai_services_resource_test.go

```diff
@@ -0,0 +1,897 @@
+		},
+	})
+}
+
+func TestAccCognitiveAzureAIServices_complete(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_azure_ai_services", "test")
+	r := AzureAIServicesResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("tags.%").HasValue("1"),
+				check.That(data.ResourceName).Key("tags.Acceptance").HasValue("Test"),
+				check.That(data.ResourceName).Key("primary_access_key").Exists(),
+				check.That(data.ResourceName).Key("secondary_access_key").Exists(),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccCognitiveAzureAIServices_update(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_azure_ai_services", "test")
+	r := AzureAIServicesResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("tags.%").HasValue("0"),
+				check.That(data.ResourceName).Key("primary_access_key").Exists(),
+				check.That(data.ResourceName).Key("secondary_access_key").Exists(),
+			),
+		},
+		{
+			Config: r.complete(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+				check.That(data.ResourceName).Key("tags.%").HasValue("1"),
+				check.That(data.ResourceName).Key("tags.Acceptance").HasValue("Test"),
+				check.That(data.ResourceName).Key("primary_access_key").Exists(),
+				check.That(data.ResourceName).Key("secondary_access_key").Exists(),
+			),
+		},
+	})
+}
+
+func TestAccCognitiveAzureAIServices_networkACLsVirtualNetworkRules(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_azure_ai_services", "test")
+	r := AzureAIServicesResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.networkACLsVirtualNetworkRules(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.networkACLsVirtualNetworkRulesUpdated(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccCognitiveAzureAIServices_networkACLs(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_azure_ai_services", "test")
+	r := AzureAIServicesResource{}
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.networkACLs(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.networkACLsUpdated(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
```

Comment:
It's good to be thorough but I don't see a benefit in splitting the tests for updating the `network_acls`, we can test the updates for all the fields and sub-blocks in `network_acls` in one test instead of separating them. Please combine these into one test.